src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8481,5 +8428,418 @@ void I.M2()
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : IInterface<string?>
+{{
+    public string? MethodAllowNull(string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    public string MethodDisallowNull([DisallowNull] string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+}}");
+        }
+
+        [WorkItem(39256, "https://github.com/dotnet/roslyn/issues/39256")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnalysisAttributeOnNonNullableReferenceTypeByRef()
+        {
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    ref T MethodMaybeNull([MaybeNull] ref T value);
+
+    [return: NotNull]
+    ref T MethodNotNull([NotNull] ref T value);
+}}
+
+class Class : [|IInterface<string>|]
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    ref T MethodMaybeNull([MaybeNull] ref T value);
+
+    [return: NotNull]
+    ref T MethodNotNull([NotNull] ref T value);
+}}
+
+class Class : IInterface<string>
+{{
+    public ref string? MethodMaybeNull(ref string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    public ref string MethodNotNull(ref string value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+}}");
+        }
+
+        [WorkItem(39256, "https://github.com/dotnet/roslyn/issues/39256")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnalysisAttributeOnNullableReferenceTypeByRef()
+        {
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    ref T MethodMaybeNull([MaybeNull] ref T value);
+
+    [return: NotNull]
+    ref T MethodNotNull([NotNull] ref T value);
+}}
+
+class Class : [|IInterface<string?>|]
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    ref T MethodMaybeNull([MaybeNull] ref T value);
+
+    [return: NotNull]
+    ref T MethodNotNull([NotNull] ref T value);
+}}
+
+class Class : IInterface<string?>
+{{
+    public ref string? MethodMaybeNull(ref string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    public ref string MethodNotNull([NotNull] ref string? value)
```

Comment:
⚠️ We should verify this one with the compiler as well. It might need to have the attribute form of the return value.

```suggestion
    [return: NotNull]
    public ref string? MethodNotNull([NotNull] ref string? value)
```